### PR TITLE
Add configurable extension support

### PR DESCRIPTION
**Pull request by dignifiedquire**
Depends on #32 

Needed for https://github.com/ipfs/js-ipfs-ipld

cc @diasdavid

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -4,11 +4,13 @@
...
-    "test": "npm run test:node && npm run test:browser",
-    "test:node": "mocha tests/node-tests.js",
-    "test:browser": "karma start karma.conf.js",
-    "coverage": "istanbul cover --print both -- _mocha tests/node-tests.js",
-    "lint": "standard"
+    "test": "dignified-test",
+    "test:node": "dignified-test node",
+    "test:browser": "dignified-test browser",
+    "build": "dignified-build",
+    "coverage": "istanbul cover --print both -- _mocha test/node.js",
```
Comment (diasdavid) [inline package.json]: Could you update this one to be `dignified-coverage`

File: src/stores/config.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ exports.setUp = (basePath, blobStore, locks) => {
           let result
           try {
             result = JSON.parse(config.toString())
-          } catch (err) {
-            return callback(err)
+          } catch (error) {
```
Comment (diasdavid) [inline src/stores/config.js]: ?

File: test/repo-test.js [hunk 3]
```diff
@@ -10,9 +10,13 @@ const fs = require('fs')
 const isNode = !global.window
 
 const fileA = isNode
-  ? fs.readFileSync(process.cwd() + '/tests/test-repo/blocks/12207028/122070286b9afa6620a66f715c7020d68af3d10e1a497971629c07606bfdb812303d.data')
+  ? fs.readFileSync(process.cwd() + '/test/test-repo/blocks/12207028/122070286b9afa6620a66f715c7020d68af3d10e1a497971629c07606bfdb812303d.data')
   : require('buffer!./test-repo/blocks/12207028/122070286b9afa6620a66f715c7020d68af3d10e1a497971629c07606bfdb812303d.data')
 
+const fileAExt = isNode
+  ? fs.readFileSync(process.cwd() + '/test/test-repo/blocks/12207028/122070286b9afa6620a66f715c7020d68af3d10e1a497971629c07606bfdb812303d.ext')
+  : require('buffer!./test-repo/blocks/12207028/122070286b9afa6620a66f715c7020d68af3d10e1a497971629c07606bfdb812303d.ext')
```
Comment (diasdavid) [inline test/repo-test.js]: thought we didn't need this anymore since dignified uses brfs. Could you confirm pls?

**Comment by diasdavid** [comment]
Overall looks good. I see a lot of code changes that are more codestyle nittpicks than features, its ok.

Just check if we still need the buffer loaders and good to be merged.

**Comment by dignifiedquire** [comment]
@diasdavid done

**Comment by diasdavid** [comment]
"go home circle-ci, you are drunk"

**Comment by dignifiedquire** [comment]
:tada:

**Pull request merged**